dvc/tree/gdrive.py

```diff
@@ -510,20 +532,41 @@ def _path_to_item_ids(self, path, create, use_cache):
         )
 
     def _get_item_id(self, path_info, create=False, use_cache=True, hint=None):
-        assert path_info.bucket == self._bucket
+        # cache is not used when Tree is associated with a GDriveDependency
+        if not self.path_info:
+            use_cache = False
 
-        item_ids = self._path_to_item_ids(path_info.path, create, use_cache)
+        item_ids = self._path_to_item_ids(
+            path_info.path, create, use_cache, path_info.bucket
+        )
         if item_ids:
             return min(item_ids)
 
         assert not create
         raise FileMissingError(path_info, hint)
 
     def exists(self, path_info, use_dvcignore=True):
+        from pydrive2.files import ApiRequestError
+
+        assert isinstance(path_info, GDriveURLInfo)
         try:
-            self._get_item_id(path_info)
+            item_id = self._get_item_id(path_info)
+            file_repr = self._drive.CreateFile({"id": item_id})
+            file_repr.FetchMetadata()
+
         except FileMissingError:
             return False
+        except ApiRequestError as e:
```

Comment:
could you please clarify, when does it happen? (why `FileMissingError` is not enough). It feels that it would be better to encapsulate this into the function that raises the `FileMissingError`, wdyt?

Reply:
When the url has no path (`gdrive://ID`) it doesn't raise `FileMissingError`. I agree it's a good idea to encapsulate in the same function that raises the `FileMissingError`. However, the way it's written, the change will affect other function and may seem too complicated for the problem it is solving, so I tried something simpler. I'll propose an encapsulated solution now.

Reply:
Done!